build/bind_artifacts.jl

```diff
@@ -25,37 +23,24 @@ function sha256sum(tarball_path)
     end
 end
 
-function get_release_asset_urls()
-    io = IOBuffer()
-    Downloads.download(ASSETS_URL, io)
-    json = JSON3.read(String(take!(io)))
-    return unescapeuri.(get.(json[:assets], :browser_download_url))
-end
-
 if abspath(PROGRAM_FILE) == @__FILE__
-    @info "Fetching assets for $TAG"
-    artifacts_urls = get_release_asset_urls()
+    # Start with a clean Artifacts.toml so that unsupported platforms are removed
+    isfile(ARTIFACTS_TOML) && rm(ARTIFACTS_TOML)
 
-    for artifact_url in artifacts_urls
-        @info "Adding artifact for $(basename(artifact_url))"
-        artifact_path = joinpath(DIR, basename(artifact_url))
-        Downloads.download(artifact_url, artifact_path)
-
-        m = match(TARBALL_REGEX, basename(artifact_path))
-        if isnothing(m)
-            throw(ArgumentError("Could not parse host triplet from $(basename(artifact_path))"))
-        end
+    for platform in REQUIRED_PLATFORMS
```

Comment:
By using a list of required platforms we can ensure that if we failed to upload all of the artifacts that the `bind_artifacts.jl` script will fail as it will attempt to download a file that doesn't exist